staging/src/k8s.io/kubectl/pkg/cmd/logs/logs_nodes.go

```diff
@@ -0,0 +1,585 @@
+	}
+	if logsOptions.TailLines != nil && *logsOptions.TailLines < -1 {
+		return fmt.Errorf("--tail must be greater than or equal to -1")
+	}
+	if o.BootChanged && (o.Boot < -100 || o.Boot > 0) {
+		return fmt.Errorf("--boot accepts values [-100, 0]")
+	}
+
+	return nil
+}
+
+// RunLogs retrieves node logs
+func (o NodeLogsOptions) RunLogs() error {
+	requests, err := o.LogsForObject(o.RESTClientGetter, o.Object, o.Options, o.GetNodeTimeout, false)
+	if err != nil {
+		return err
+	}
+
+	// buffer output for slightly better streaming performance
+	out := bufio.NewWriterSize(o.Out, 1024*16)
+	defer out.Flush()
+
+	if o.Unify {
+		// unified output is each source, interleaved in lexographic order (assumes
+		// the source input is sorted by time)
+		return o.parallelConsumeRequest(requests)
+	}
+
+	return o.sequentialConsumeRequest(requests)
+}
+
+func (o NodeLogsOptions) parallelConsumeRequest(requests map[corev1.ObjectReference]rest.ResponseWrapper) error {
+	o.Prefix = false
+	wg := &sync.WaitGroup{}
+	wg.Add(len(requests))
+	var readers []Reader
+	for objRef, request := range requests {
+		reader, writer := io.Pipe()
+		readers = append(readers, Reader{
+			R: reader,
+		})
+		go func(objRef corev1.ObjectReference, request rest.ResponseWrapper) {
+			defer wg.Done()
+			if err := o.ConsumeRequestFn(request, writer); err != nil {
+				fmt.Fprintf(writer, "error: %v\n", err)
+				writer.CloseWithError(err)
+				// It's important to return here to propagate the error via the pipe
+				return
+			}
+		}(objRef, request)
+
+		go func() {
+			wg.Wait()
+			writer.Close()
+		}()
+	}
+
+	_, err := NewMergeReader(readers...).WriteTo(o.Out)
+
+	return err
+}
+
+func (o NodeLogsOptions) sequentialConsumeRequest(requests map[corev1.ObjectReference]rest.ResponseWrapper) error {
+	for _, request := range requests {
+		if err := o.ConsumeRequestFn(request, o.Out); err != nil {
+			fmt.Fprintf(o.Out, "error: %v\n", err)
+			return err
+		}
+	}
+
+	return nil
+}
+
+// NodeLogsConsumeRequest reads the data from request and writes into
+// the out writer.
+func (o NodeLogsOptions) NodeLogsConsumeRequest(request rest.ResponseWrapper, out io.Writer) error {
+	in, err := request.Stream(context.TODO())
+	if err != nil {
+		return err
+	}
+	defer in.Close()
+
+	// raw output implies we may be getting binary content directly
+	// from the remote and so we want to perform no translation
+	if o.Raw {
+		// TODO: optionallyDecompress should be implemented by checking
+		// the content-encoding of the response, but we perform optional
+		// decompression here in case the content of the logs on the server
+		// is also gzipped.
+		return optionallyDecompress(out, in)
+	}
+
+	var prefix []byte
+	if o.Prefix {
+		prefix = []byte(fmt.Sprintf("%s ", o.Node))
+	}
+
+	return outputDirectoryEntriesOrContent(out, in, prefix)
+}
+
+func optionallyDecompress(out io.Writer, in io.Reader) error {
+	bufferSize := 4096
```

Comment:
should be a constant, same with line 407

Reply:
done